lib/components/intent.js

```diff
@@ -97,6 +103,34 @@ function Intent(client, botClient, opts) {
         }
     }
 
+    if (!opts.caching) {
+        opts.caching = {
+
+        }
+    }
+
+    opts.caching.ttl = opts.caching.ttl === undefined ? DEFAULT_CACHE_TTL : opts.caching.ttl;
+    opts.caching.size = opts.caching.size === undefined ? DEFAULT_CACHE_SIZE : opts.caching.ttl;
+    this._requestCaches = {};
+    this._requestCaches.profile = new ClientRequestCache(
+        opts.caching.ttl,
+        opts.caching.size,
+        (_, userId, info) => {
+        return this.intent.getProfileInfo(userId, info, false);
```

Comment:
indentation (on this and the 2 below).

I'd also recommend having the closing curly not be part of the closing parens